### DISCUSSION TITLE
How to actually run any bevy example using OpenGL?

### DISCUSSION THREAD

**Discussion by dmitrykolesnikovich** [bevyengine/bevy#11568] [Q&A]
I tried to change VULKAN to GL in window.rs file but without effect

![image](https://github.com/bevyengine/bevy/assets/6131040/a2833fbf-8a2c-4efa-9935-dc98c2a8253d)

**Comment by st0rmbtw** [comment]
Hi! That's not how you do it. You can change the backend by setting the `backends` field of the `WgpuSettings` in the `RenderPlugin`.
Here's the example:
https://github.com/bevyengine/bevy/blob/56c3079586e9747183dad034b39514a46fbe4b03/examples/app/no_renderer.rs#L17

**Comment by JMS55** [comment]
You can also set the `WGPU_BACKEND=GL` (iirc it's GL) environment variable. Keep in mind, I would not use opengl unless you have to. Not all features will be supported.

**Comment by dmitrykolesnikovich** [comment]
It seems like GL not supported

```
Unable to find a GPU! Make sure you have installed required drivers!
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

**Reply by eira-fransham** [reply]
Can confirm that this still happens for me

**Comment by Johannes0021** [comment]
I think to use OpenGL:

Enable the `gles` feature in the `bevy_render` crate.
```toml
bevy_render = { version = "0.19.1", features = ["gles"] }
```

Enable it in the code:
```rust
DefaultPlugins.set(RenderPlugin {
    render_creation: RenderCreation::Automatic(Box::new(WgpuSettings {
        priority: WgpuSettingsPriority::WebGL2,
        limits: WgpuLimits::downlevel_webgl2_defaults()
            .using_resolution(WgpuLimits::default()),
        backends: Some(Backends::GL),
        ..default()
    })),
    ..default()
})
```

Maybe also `.disable::<PbrPlugin>())`
